code/hash/dig-vs-[]-vs-fetch.rb

```diff
@@ -0,0 +1,27 @@
+require 'benchmark/ips'
+
+h = { a: { b: { c: { d: { e: "foo" } } } } }
+
+Benchmark.ips do |x|
+  x.report 'Hash#dig' do
+    h.dig(:a, :b, :c, :d, :e)
+  end
+
+  x.report 'Hash#[]' do
+    h[:a][:b][:c][:d][:e]
+  end
+
+  x.report 'Hash#[] fallback' do
+    ((((h[:a] || {})[:b] || {})[:c] || {})[:d] || {})[:e]
```

Comment:
Actually, I think, the way proposed by @nateberkopec is more effective in case when none of the keys are exists, because you don't need to create all of these hashes.